tests/Doctrine/Tests/ORM/Functional/Ticket/GH10049/SetInheritedReadOnlyPropertyValueTest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket\GH10049;
+
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @requires PHP 8.1
+ */
+class SetInheritedReadOnlyPropertyValueTest extends OrmFunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(
+            ReadOnlyPropertyOwner::class,
+            ReadOnlyPropertyInheritor::class
+        );
+    }
+
+    public function test(): void
+    {
+        $child = new ReadOnlyPropertyInheritor(10049);
+        $this->_em->persist($child);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $entity = $this->_em->find(ReadOnlyPropertyInheritor::class, 10049);
+
+        self::assertInstanceOf(ReadOnlyPropertyInheritor::class, $entity);
+        self::assertSame(10049, $entity->id);
```

Comment:
I think you added these assertions to bypass the warning, but what really matters the most to you here is that this test executes without a crash. I'd remove both assertions and use `@doesNotPerformAssertions` (https://docs.phpunit.de/en/10.1/annotations.html?highlight=doesNotPerformAssertions#doesnotperformassertions), it's less misleading.